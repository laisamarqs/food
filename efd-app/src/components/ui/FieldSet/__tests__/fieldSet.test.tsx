import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import { FieldSet } from '..'
import { expect } from 'vitest'

describe('<FieldSet />', () => {
    it('should render correctly', () => {
      const { container } = render(
        <FieldSet.Root>
          <FieldSet.Label id="name" label="Nome" />
          <FieldSet.Input id="name" required="true" type="text" />
        </FieldSet.Root>
      )
  
      expect(screen.getByText('Nome')).toBeInTheDocument()
  
      expect(container.firstChild).toMatchSnapshot()
    })
  
    it('should render correctly with input mask', () => {
      render(
        <FieldSet.Root>
          <FieldSet.Label id="month" label="Mês" />
          <FieldSet.InputMask mask="9-9" id="month" required="true" type="text" />
        </FieldSet.Root>
      )
  
      const inputMask = screen.getByRole('textbox', { name: 'Mês' })
  
      fireEvent.change(inputMask, { target: { value: '00' } })
  
      expect(inputMask).toHaveValue('0-0')
    })
    it('should render an error message when receiving a error', () => {
      render(
        <FieldSet.Root>
          <FieldSet.Label id="month" label="Mês" />
          <FieldSet.InputMask mask="9-9" id="month" required="true" type="text" />
          <FieldSet.ErrorMessage errorMessage="O campo é obrigatório" />
        </FieldSet.Root>
      )
  
      expect(screen.getByText('O campo é obrigatório')).toBeInTheDocument()
    })
    it('should correctly receive the change event', () => {
      render(
        <FieldSet.Root>
          <FieldSet.Label id="name" label="Nome" />
          <FieldSet.Input id="name" required="true" type="text" />
        </FieldSet.Root>
      )
  
      const input = screen.getByRole('textbox', { name: 'Nome' })
  
      fireEvent.change(input, { target: { value: 'Ana' } })
  
      expect(input).toHaveValue('Ana')
    })
  })