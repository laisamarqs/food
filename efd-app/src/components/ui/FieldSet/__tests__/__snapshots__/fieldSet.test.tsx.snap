// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<FieldSet /> > should render correctly 1`] = `
<div
  aria-label="Campo do FormulÃ¡rio"
  class="sc-dcJsrY bAxpnH"
>
  <label
    for="name"
  >
    Nome
  </label>
  <input
    aria-required="true"
    id="name"
    name="name"
    type="text"
    value=""
  />
</div>
`;